src/main/java/io/appium/java_client/AppiumFluentWait.java

```diff
@@ -222,32 +245,46 @@ public <V> V until(Function<? super T, V> isTrue) {
             // Check the timeout after evaluating the function to ensure conditions
             // with a zero timeout can succeed.
             if (end.isBefore(getClock().instant())) {
-                String message = getMessageSupplier() != null ? getMessageSupplier().get() : null;
-
-                String timeoutMessage = String.format(
-                        "Expected condition failed: %s (tried for %d second(s) with %s interval)",
-                        message == null ? "waiting for " + isTrue : message,
-                        getTimeout().getSeconds(), getInterval());
-                throw timeoutException(timeoutMessage, lastException);
+                handleTimeoutException(lastException, isTrue);
             }
 
-            try {
-                Duration interval = getInterval();
-                if (pollingStrategy != null) {
-                    final IterationInfo info = new IterationInfo(iterationNumber,
-                            Duration.between(start, getClock().instant()), getTimeout(),
-                            interval);
-                    interval = pollingStrategy.apply(info);
-                }
-                getSleeper().sleep(interval);
-            } catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
-                throw new WebDriverException(e);
-            }
+            Duration interval = getIntervalWithPollingStrategy(start, iterationNumber);
+            sleepWaitFor(interval);
+
             ++iterationNumber;
         }
     }
 
+    private <V> void handleTimeoutException(Throwable lastException, Function<? super T, V> isTrue) {
+        String message = getMessageSupplier() != null ? getMessageSupplier().get() : null;
+
+        String timeoutMessage = String.format(
+                "Expected condition failed: %s (tried for %d second(s) with %s interval)",
+                message == null ? "waiting for " + isTrue : message,
+                getTimeout().getSeconds(), getInterval());
```

Comment:
consider transforming interval into some explicit time unit. The default string representation of this class is not very readable

Reply:
For now, I just did a refactor to make the big method smaller based on the specific responsibility. But I see your point, and I'll make the changes.

Reply:
It was tricky because if I choose a TimeUnit and the timeout/interval is less than 1, it returns 0. 

A possible solution is to print the duration directly, but it returns the ISO 8601 format, and it is not a user-friendly value, i.e., `PT1S`.

Milliseconds could be a compromise. I consider it improbable to have a timeout/interval less than 1 milliseconds. What do you think?

Reply:
yes, milliseconds would be fine